File: core/src/main/java/pl/alpheratzteam/obfuscator/transformer/optimizers/NopRemoverTransformer.java
package pl.alpheratzteam.obfuscator.transformer.optimizers;

import org.objectweb.asm.tree.ClassNode;
import pl.alpheratzteam.obfuscator.Obfuscator;
import pl.alpheratzteam.obfuscator.transformer.Transformer;
import java.util.Arrays;
import java.util.Map;

public class NopRemoverTransformer extends Transformer
{
    public NopRemoverTransformer(Obfuscator obfuscator) {
        super(obfuscator);
    }

    @Override
    public void visit(Map<String, ClassNode> classMap) {
        classMap.values().forEach(classNode ->
                classNode.methods.forEach(methodNode ->
                        Arrays.stream(methodNode.instructions.toArray())
                                .filter(insnNode -> insnNode.getOpcode() == NOP)
                                .forEach(insnNode -> methodNode.instructions.remove(insnNode))));
    }
}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/transformer/optimizer/NopRemoverTransformer.kt
package pl.alpheratzteam.obfuscator.transformer.optimizer

import org.objectweb.asm.Opcodes.NOP
import pl.alpheratzteam.obfuscator.Obfuscator
import pl.alpheratzteam.obfuscator.api.transformer.Transformer

/**
 * @author Unix
 * @since 18.12.2020
 */

class NopRemoverTransformer : Transformer {
    override fun transform(obfuscator: Obfuscator) = obfuscator.classes.values.forEach { it.methods.forEach { methodNode -> methodNode.instructions.filter { it.opcode == NOP }.forEach { methodNode.instructions.remove(it) } } }
}